source/2_platform/rex_directx/src/directx_renderer.cpp

```diff
@@ -0,0 +1,137 @@
+#include "rex_directx/directx_feature_level.h"
+#include "rex_directx/directx_feature_shader_model.h"
+#include "rex_directx/directx_util.h"
+#include "rex_directx/dxgi/adapter.h"
+#include "rex_directx/dxgi/adapter_manager.h"
+#include "rex_directx/dxgi/factory.h"
+#include "rex_directx/log.h"
+#include "rex_renderer_core/renderer.h"
+#include "rex_std/memory.h"
+#include "rex_std_extra/utility/enum_reflection.h"
+
+namespace rex
+{
+  namespace renderer
+  {
+    namespace internal
+    {
+      //-------------------------------------------------------------------------
+      const Gpu* highest_scoring_gpu(const rsl::vector<rsl::unique_ptr<Gpu>>& gpus)
+      {
+        auto it = std::max_element(gpus.cbegin(), gpus.cend(),
+                                   [](const rsl::unique_ptr<Gpu>& lhs, const rsl::unique_ptr<Gpu>& rhs)
+                                   {
+                                     const size_t lhs_vram = lhs->description().dedicated_video_memory.size_in_bytes();
+                                     const size_t rhs_vram = rhs->description().dedicated_video_memory.size_in_bytes();
+
+                                     return rhs_vram > lhs_vram;
+                                   });
+
+        return it != gpus.cend() ? it->get() : nullptr;
+      }
+    } // namespace internal
+
+    namespace directx
+    {
+      RendererInfo g_renderer_info; // NOLINT (fuchsia-statically-constructed-objects, cppcoreguidelines-avoid-non-const-global-variables)
+    }                               // namespace directx
+
+    //-------------------------------------------------------------------------
+    const RendererInfo& info()
+    {
+      return directx::g_renderer_info;
+    }
+
+    //-------------------------------------------------------------------------
+    ShaderPlatform shader_platform()
+    {
+      return ShaderPlatform::HLSL;
+    }
+
+    //-------------------------------------------------------------------------
+    bool is_y_up()
+    {
+      return false;
+    }
+
+    //-------------------------------------------------------------------------
+    bool is_depth_0_to_1()
+    {
+      return true;
+    }
+
+    namespace backend
+    {
+      struct DirectXContext
```

Comment:
just letting you know you have a win_directx_context.cpp file in RexWindows, but the DirectXContext is defined here, not sure if that's what you wanted

Reply:
Yes I know but this is actually a struct containing device dependent objects, I called it context cuz of lack of better naming